### PR TITLE
[FHIR] added workflow to lint and test charts

### LINKED ISSUE #36: [FHIR] Add chart-testing workflow
Per https://github.com/Alvearie/alvearie-helm/issues/15, chart-testing allows us to statically lint the chart ensuring Chart.yaml and values.yaml are valid according to our configuration. It also allows us to install the chart to an actual k8s cluster as part of the CI.

### PR DESCRIPTION
**Pull request by chgl**
Closes #36

Please note that this includes 3 changes which were arguably outside the scope of just adding a new workflow:

1. I had to reduce the default replicaCount to 1 because the hosted GitHub runners only have 7GiB of RAM by default (https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners). Note that alternatively we could specify custom `values.yaml` files to be used by chart-testing if we want to keep the default at 2 replicas (see https://github.com/helm/chart-testing/blob/main/doc/ct_install.md). Or we could remove the resource requests entirely. Or reduce the replica count to 1 by default **and** remove the resource requests/limits - which is maybe a bit more common among the charts found in the wild.

1. I added the `home` and `type` fields to the Chart.yaml - mostly because the chart-schema.yaml I copy-pasted had them set as required. The former adds a `Homepage` to the list of links on artifacthub - which may be nice, the latter is just for completeness sake.

1. I made the helm test job a bit more fault-tolerant by adding retries and increased timeouts. This should help avoiding flaky test behavior - this is related to the server becoming "ready" before the schema job has completed, see https://github.com/Alvearie/alvearie-helm/pull/28#issuecomment-918558104

### CODE REVIEW

**Comment by chgl** [comment]
Because I had forgotten to update the README.md using helm-docs in my original commit, I just added a new job to run pre-commit (and therefore helm-docs) as part of the CI as well.

Installation was a bit more involved since the https://github.com/pre-commit/action action is unfortunately deprecated.

**Comment by lmsurpre** [comment]
> I had to reduce the default replicaCount to 1 because the hosted GitHub runners only have 7GiB of RAM by default (https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners). Note that alternatively we could specify custom values.yaml files to be used by chart-testing if we want to keep the default at 2 replicas (see https://github.com/helm/chart-testing/blob/main/doc/ct_install.md). Or we could remove the resource requests entirely. Or reduce the replica count to 1 by default and remove the resource requests/limits - which is maybe a bit more common among the charts found in the wild.

My thoughts:
1. I prefer a replica count of 2 because our server is stateless (minus caching for performance) and I think its best practice to have more than one replica. That said, I'm not totally opposed to dropping it to 1 if it gets us a running CI.
2. I want to remove the resource requests from the deployment, but I think we have some work to do there to make that possible.  For historical reasons, the FHIR server defaults to a min and max heap size of 4GB.  It does run with much less though.  We have tasks under ibm.com/fhir for better understanding on our memory usage and that could inform our defaults.  Til then, it is totally possible to override the default min/max heap size of liberty by placing a jvm.options file under configDropins/overrides.  I'll open a separate task for adding that to this chart.

File: charts/ibm-fhir-server/values.yaml [hunk 1]
```diff
@@ -17,6 +17,7 @@ restrictEndpoints: false
 # -- Control which interactions are supported for which resource type endpoints
 # @default -- A single entry for resourceType "Resource" that applies to all resource types
 endpoints:
+  # yamllint disable-line rule:comments
```
Comment (lmsurpre) [inline charts/ibm-fhir-server/values.yaml:20]: what does this one do?

File: .github/workflows/lint-and-test.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@v2.1.0
+
+      - name: Run chart-testing (list-changed)
+        id: list-changed
+        run: |
+          changed=$(ct list-changed --config .ct/config.yaml)
+          if [[ -n "$changed" ]]; then
+            echo "::set-output name=changed::true"
+          fi
```
Comment (lmsurpre) [inline .github/workflows/lint-and-test.yaml:70]: this looked a little odd to me, but it does do basically exactly what they suggest at https://github.com/helm/chart-testing-action so I think its good

File: charts/ibm-fhir-server/values.yaml (hunk 1 above)
Comment (chgl) [reply charts/ibm-fhir-server/values.yaml:20]: the default yamllint config requires comments to each be on their own line. Without this inline comment, chart-testing lint and yamllint would complain about:

```console
\charts\ibm-fhir-server\values.yaml
  20:5      warning  too few spaces before comment  (comments)
```

Doing:

```yaml
endpoints:
  -
    # -- A valid FHIR resource type; use "Resource" for whole-system behavior
    resourceType: Resource
```

would work, but be reverted by prettier... So just disabling the yamllint rule for this particular case seemed the cleanest.

**Comment by chgl** [comment]
> My thoughts:
> 
> 1. I prefer a replica count of 2 because our server is stateless (minus caching for performance) and I think its best practice to have more than one replica. That said, I'm not totally opposed to dropping it to 1 if it gets us a running CI.

I completely agree, for any production usage, there should be more than 1 replica. But I also think a default of 1 is reasonable. I've added https://github.com/Alvearie/alvearie-helm/issues/39 which is helpful in >1 replica scenarios.

> 2. I want to remove the resource requests from the deployment, but I think we have some work to do there to make that possible.  For historical reasons, the FHIR server defaults to a min and max heap size of 4GB.  It does run with much less though.  We have tasks under ibm.com/fhir for better understanding on our memory usage and that could inform our defaults.  Til then, it is totally possible to override the default min/max heap size of liberty by placing a jvm.options file under configDropins/overrides.  I've opened [Support mix/max heap size overrides #38](https://github.com/Alvearie/alvearie-helm/issues/38) for that.

Ah I see, thank you! This may be off-topic, but does setting `XX:MaxRAMPercentage` (https://www.atamanroman.dev/articles/usecontainersupport-to-the-rescue/) work? So we'd only have to specify the percentage once and then have it use the container resource limits.

File: charts/ibm-fhir-server/values.yaml (hunk 1 above)
Comment (lmsurpre) [reply charts/ibm-fhir-server/values.yaml:20]: ok, thanks for the explainer

**Review by lmsurpre — APPROVED** [review]

**Review by michaelwschroeder — APPROVED** [review]

**Pull request merged**